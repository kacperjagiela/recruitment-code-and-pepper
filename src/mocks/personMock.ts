import { Person } from '~/types';

const personMock = {
    name: 'C-3PO',
    height: 167,
    mass: 75,
    hair_color: 'n/a',
    skin_color: 'gold',
    eye_color: 'yellow',
    birth_year: '112BBY',
    gender: 'n/a',
} as Person;

export default personMock;
