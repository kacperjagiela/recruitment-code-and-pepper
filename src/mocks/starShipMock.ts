import { StarShip } from '~/types';

const starShipMock = {
    name: 'CR90 corvette',
    model: 'CR90 corvette',
    manufacturer: 'Corellian Engineering Corporation',
    cost_in_credits: 3500000,
    length: 150,
    max_atmosphering_speed: 950,
    crew: 30,
    passengers: 600,
    cargo_capacity: 3000000,
    consumables: { days: 0, months: 0, years: 1, fullString: '1 year' },
    hyperdrive_rating: 2.0,
    MGLT: 60,
} as StarShip;

export default starShipMock;
