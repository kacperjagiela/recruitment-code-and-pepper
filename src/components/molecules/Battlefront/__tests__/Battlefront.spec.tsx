import { render } from '@testing-library/react';

import personMock, { secondPersonMock } from '~/mocks/personMock';
import starShipMock from '~/mocks/starShipMock';

import { Battlefront } from '..';

describe('<Battlefront />', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Battlefront
                firstDuelist={personMock}
                secondDuelist={secondPersonMock}
                currentAttribute="mass"
                currentBattleType="person"
            />,
        );

        expect(container).toMatchSnapshot();
    });

    it('renders correctly for starships', () => {
        const { container } = render(
            <Battlefront
                firstDuelist={starShipMock}
                secondDuelist={starShipMock}
                currentAttribute="length"
                currentBattleType="starships"
            />,
        );

        expect(container).toMatchSnapshot();
    });
});
