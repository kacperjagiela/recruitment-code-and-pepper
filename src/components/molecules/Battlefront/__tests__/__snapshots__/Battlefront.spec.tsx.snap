// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Battlefront /> renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1bhcfiz"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1h4pobq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            name:
          </span>
           
          <b>
            C-3PO
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            height:
          </span>
           
          <b>
            167
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1sablf2-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            mass:
          </span>
           
          <b>
            75
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            hair color:
          </span>
           
          <b>
            n/a
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            skin color:
          </span>
           
          <b>
            gold
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            eye color:
          </span>
           
          <b>
            yellow
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            birth year:
          </span>
           
          <b>
            112BBY
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            gender:
          </span>
           
          <b>
            n/a
          </b>
        </p>
      </div>
    </div>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-sd46zu-MuiTypography-root"
    >
      VS
    </h5>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-toqqfd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            name:
          </span>
           
          <b>
            C-3PO1
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            height:
          </span>
           
          <b>
            167
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1sablf2-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            mass:
          </span>
           
          <b>
            76
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            hair color:
          </span>
           
          <b>
            n/a
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            skin color:
          </span>
           
          <b>
            gold
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            eye color:
          </span>
           
          <b>
            yellow
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            birth year:
          </span>
           
          <b>
            112BBY
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            gender:
          </span>
           
          <b>
            n/a
          </b>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Battlefront /> renders correctly for starships 1`] = `
<div>
  <div
    class="MuiBox-root css-1bhcfiz"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1h4pobq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            name:
          </span>
           
          <b>
            CR90 corvette
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            model:
          </span>
           
          <b>
            CR90 corvette
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            manufacturer:
          </span>
           
          <b>
            Corellian Engineering Corporation
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            cost in credits:
          </span>
           
          <b>
            3500000
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1sablf2-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            length:
          </span>
           
          <b>
            150
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            max atmosphering speed:
          </span>
           
          <b>
            950
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            crew:
          </span>
           
          <b>
            30
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            passengers:
          </span>
           
          <b>
            600
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            cargo capacity:
          </span>
           
          <b>
            3000000
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            consumables:
          </span>
           
          <b>
            1 year
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            hyperdrive rating:
          </span>
           
          <b>
            2
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            MGLT:
          </span>
           
          <b>
            60
          </b>
        </p>
      </div>
    </div>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-sd46zu-MuiTypography-root"
    >
      VS
    </h5>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1h4pobq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            name:
          </span>
           
          <b>
            CR90 corvette
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            model:
          </span>
           
          <b>
            CR90 corvette
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            manufacturer:
          </span>
           
          <b>
            Corellian Engineering Corporation
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            cost in credits:
          </span>
           
          <b>
            3500000
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1sablf2-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            length:
          </span>
           
          <b>
            150
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            max atmosphering speed:
          </span>
           
          <b>
            950
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            crew:
          </span>
           
          <b>
            30
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            passengers:
          </span>
           
          <b>
            600
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            cargo capacity:
          </span>
           
          <b>
            3000000
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            consumables:
          </span>
           
          <b>
            1 year
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            hyperdrive rating:
          </span>
           
          <b>
            2
          </b>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1269381-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
          >
            MGLT:
          </span>
           
          <b>
            60
          </b>
        </p>
      </div>
    </div>
  </div>
</div>
`;
